File: coffee-order-api/models/voucher.js
const mongoose = require("mongoose");
const generateUUID = require("../utils/crypto/generateId");
const User = require("./user");

const voucherSchema = new mongoose.Schema({
  _id: {
    type: String,
    default: generateUUID,
  },
  type: {
    type: String,
    enum: ["Discount", "FreeCoffee"],
    required: true,
  },
  client: {
    type: mongoose.Schema.Types.String,
    ref: "User",
    autopopulate: { select: "-password -publicKey" },
  },
  used: {
    type: Boolean,
    default: false,
  },
  date: {
    type: Date,
    default: Date.now,
    required: true,
  },
});

voucherSchema.plugin(require("mongoose-autopopulate"));

module.exports = mongoose.model("Voucher", voucherSchema);

File: coffee-order-application/app/src/main/java/com/feup/coffee_order_application/domain/model/Voucher.kt
package com.feup.coffee_order_application.domain.model

data class Voucher(
    val _id: String,
    val type: String,
    val client: User,
    val used: Boolean,
    val date: String,
    var isSelected: Boolean = false
){
    companion object {
        const val TYPE_DISCOUNT = "Discount"
        const val TYPE_FREE_COFFEE = "FreeCoffee"
    }
}
